src/gluonts/dataset/split/splitter.py

```diff
@@ -266,7 +263,28 @@ def _check_split_length(
     assert train_end + prediction_length <= test_end, msg
 
 
-class OffsetSplitter(pydantic.BaseModel, AbstractBaseSplitter):
+@dataclass
+class TestTemplate:
+    dataset: Iterable[DataEntry]
+    splitter: AbstractBaseSplitter
+
+    def generate_instances(self, prediction_length, *args, **kwargs):
+        yield from self.splitter._generate_test_slices(
+            self.dataset, prediction_length, *args, **kwargs
+        )
```

Comment:
```suggestion
    def generate_instances(self, prediction_length: int, windows=1, distance=None, max_history=None):
        yield from self.splitter._generate_test_slices(
            self.dataset,
            prediction_length=prediction_length,
            windows=windows,
            distance=distance,
            max_history=max_history,
        )
```

Reply:
You can also add a docstring for this method, see eg https://github.com/awslabs/gluon-ts/blob/dev/src/gluonts/transform/feature.py#L217-L232